mixer/pkg/pool/goroutine.go

```diff
@@ -48,23 +54,34 @@ func (gp *GoroutinePool) Close() error {
 	return nil
 }
 
-// ScheduleWork registers the given function to be executed at some point
-func (gp *GoroutinePool) ScheduleWork(fn WorkFunc) {
+// ScheduleWork registers the given function to be executed at some point. The given param will
+// be supplied to the function during execution.
+//
+// By making use of the supplied parameter, it is possible to avoid allocation costs when scheduling the
+// work function. The caller needs to make sure that function fn only depends on external data through the
+// passed in param interface{} and nothing else.
```

Comment:
You might consider adding that:

A way to ensure this condition is met, is by passing ScheduleWork a normal named function rather that an inline anonymous function. It's easy for code in an anonymous function to have (or gain) an accidental reference that causes a closure to silently be created.